nd4j/nd4j-common/src/main/java/org/nd4j/linalg/primitives/Atomic.java

```diff
@@ -0,0 +1,123 @@
+/*******************************************************************************
+ * Copyright (c) 2015-2018 Skymind, Inc.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+package org.nd4j.linalg.primitives;
+
+import lombok.NoArgsConstructor;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Objects;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+/**
+ *
+ * @param <T>
+ */
+@NoArgsConstructor
+public class Atomic<T extends Serializable> implements Serializable {
+    private T value;
```

Comment:
value should be transient

Reply:
How comes? The whole point - T extends Serializable

Reply:
Ah, I meant volatile